src/transformers/modeling_utils.py

```diff
@@ -1060,7 +1058,9 @@ def _generate_beam_search(
         done = [False for _ in range(batch_size)]
 
         while cur_len < max_length:
-            model_inputs = self.prepare_inputs_for_generation(input_ids, past=past, encoder_inputs=encoder_inputs, attention_mask=attention_mask)
+            model_inputs = self.prepare_inputs_for_generation(
```

Comment:
this requiring `encoder_inputs` is very hacky on my part. I would love to get rid of it.
(we can coordinate who does what/where on slack!)

Reply:
Yeah maybe there might be a better way. But I feel like either we require `encoder_inputs` and set `encoder_input=None` for non seq-2-seq or we add another if-else statement. And I would prefer the first option here. Happy to discuss!